src/sentry/db/mixin.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import absolute_import
+
+import six
+
+
+from uuid import uuid4
+
+from sentry.models import OrganizationOption
+
+
+def delete_pending_deletion_option(instance, **kwargs):
+    if not issubclass(instance.__class__, PendingDeletionMixin):
+        return
+    instance.delete_pending_deletion_option()
+
+
+class PendingDeletionMixin(object):
+    _rename_fields_on_pending_delete = []
+
+    def build_pending_deletion_key(self):
+        return 'pending-delete:%s:%s' % (self.__class__, self.id)
+
+    def rename_on_pending_deletion(self, fields=None):
+        fields = fields or self._rename_fields_on_pending_delete
+        original_data = {}
+
+        for field in fields:
+            original_data[field] = getattr(self, field)
+            setattr(self, field, uuid4().hex)
```

Comment:
Will it be alarming to customers to see uuid values in the UI?

Reply:
That's a good question. Repositories have the pending delete showing up in the UI. I'm not extremely clear on what the expected behavior is there. If I were a customer, I would want it to stay consistent for as long as it shows in the UI. The problem is that `organization_id` and `name` together must be unique. @markstory would appending something to the name be less confusing?

Reply:
I think appending to the existing name is a better solution. You could append a datetime string of or a timestamp like `exampleco/hotdogs v1238163182`